Assets/MRTK/Examples/Experimental/Elastic/Scripts/WidgetElasticDemo.cs

```diff
@@ -0,0 +1,133 @@
+            SnapPoints = new float[] { },
+            SnapToEnds = false
+        };
+
+        // The elastic extent for our backplate scaling.
+        private static LinearElasticExtent scaleExtent = new LinearElasticExtent
+        {
+            MinStretch = 0.0f,
+            MaxStretch = 1.0f,
+            SnapPoints = new float[] { },
+            SnapToEnds = false
+        };
+
+        // The elastic properties of our springs.
+        private static ElasticProperties elasticProperties = new ElasticProperties
+        {
+            Mass = 0.03f,
+            HandK = 4.0f,
+            EndK = 3.0f,
+            SnapK = 1.0f,
+            Drag = 0.2f
+        };
+
+        // Allow the user to configure the enabled/disabled states of the
+        // elements controlled by our elastic systems.
+        public float inflateAngle = -170.0f;
+        public float deflateAngle = -20.0f;
+        public float inflateScale = 1.0f;
+        public float deflateScale = 0.1f;
+
+        // Is the doodad inflated or not?
+        private bool isInflated = false;
+
+        // Start is called before the first frame update
+        void Start()
+        {
+            for (var i = 0; i < flipPanels.Count; i++)
+            {
+                flipElastics.Add(new LinearElasticSystem(deflateAngle, 0.0f, flipExtent, elasticProperties));
+                flipGoals.Add(deflateAngle);
+            }
+
+            backplateElastic = new LinearElasticSystem(deflateScale, 0.0f, scaleExtent, elasticProperties);
+            backplateGoal = deflateScale;
+        }
+
+        // Update is called once per frame
+        void Update()
+        {
+            backplateTransform.localScale = new Vector3(Mathf.Clamp01(backplateElastic.ComputeIteration(backplateGoal, Time.deltaTime)), backplateTransform.localScale.y, backplateTransform.localScale.z);
+
+            for (var i = 0; i < flipElastics.Count; i++)
+            {
+                flipPanels[i].localEulerAngles = new Vector3(Mathf.Clamp(flipElastics[i].ComputeIteration(flipGoals[i], Time.deltaTime), -360, 0), flipPanels[i].localRotation.y, flipPanels[i].localRotation.z);
+            }
+        }
+
+        public void Inflate()
+        {
+            if (!isInflated)
+            {
+                isInflated = true;
+                StartCoroutine(InflateCoroutine());
+            }
+            else
+            {
+                isInflated = false;
+                StartCoroutine(DeflateCoroutine());
+            }
+        }
+
+        public IEnumerator DeflateCoroutine()
+        {
+            for (var i = 0; i < flipElastics.Count; i++)
+            {
+                flipGoals[i] = deflateAngle;
+                yield return new WaitForSeconds(0.1f);
+            }
+            backplateGoal = deflateScale;
+        }
+
+        public IEnumerator InflateCoroutine()
+        {
+            backplateGoal = inflateScale;
+            yield return new WaitForSeconds(0.1f);
```

Comment:
The intent of the waiting in this case (in both deflate and inflate) is to create a staggered inflate/deflate experience right?  If so that could be a useful comment to leave (i.e. usually having comments on awaits/yields that do so for arbitrary time periods is helpful in understanding why they're needed)

Reply:
Fixed